sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -538,7 +532,6 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
       schema = schema,
       provider = Some(provider),
       partitionColumnNames = partitioning.asPartitionColumns,
-      bucketSpec = bucketSpec,
```

Comment:
where do we restore the bucket spec from the v2 partitioning?

Reply:
here https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala#L337
seems the columns in `BucketTransform` are not added to partitionColumns.

Reply:
This is the code path that we fallback to v1 command, and `V2SessionCatalog` won't be reached. I think we need to restore the bucket spec here, similar to partition cols.